server/src/test/java/org/apache/calcite/test/ServerTest.java

```diff
@@ -532,4 +532,28 @@ static Connection connect() throws SQLException {
       }
     }
   }
+
+  @Test public void testDropWithFullyQualifiedNameWhenSchemaDoesntExist() throws Exception {
+    try (Connection c = connect();
+         Statement s = c.createStatement()) {
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "schema", "Schema");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "table", "Table");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "materialized view", "Table");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "view", "View");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "type", "Type");
+      checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(s, "function", "Function");
+    }
+  }
+
+  private void checkDropWithFullyQualifiedNameWhenSchemaDoesntExist(
+      Statement statement, String objectType, String objectTypeInErrorMessage) throws Exception {
+    try {
+      statement.execute("drop " + objectType + " s.o");
+      fail("expected error");
```

Comment:
I suggest `assertThrows` since:
1) It better structures the code (it is easier to see which block is expected to fail)
2) It has `reason` parameter. Please add proper reasoning that explains why `expected error`.

Reply:
Reworked to `assertThrows`.